packages/kolibri-tools/lib/i18n/ExtractMessages.js

```diff
@@ -0,0 +1,804 @@
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+const sortBy = require('lodash/sortBy');
+const glob = require('glob');
+const recast = require('recast');
+const traverse = require('ast-traverse');
+const get = require('lodash/get');
+const del = require('del');
+const createCsvWriter = require('csv-writer').createObjectCsvWriter;
+const vueCompiler = require('vue-template-compiler');
+const logging = require('../logging');
+
+const GLOB = '/**/*.@(vue|js)';
+const CONTEXT_LINE = '\n-- CONTEXT --\n';
+
+/*
+ * READ THIS
+ * HEY! Are you adding a product to be covered by this script? I've added some comments
+ * with the characters NP (New Project) - search the file for them for some helpful hints
+ * at what you need to do or add when making this work with another project
+ */
+
+/*
+ * Project Specific Paths & Processes
+ *
+ * Supported Projects
+ * - Kolibri (https://github.com/learningequality/kolibri)
+ * - Studio (https://github.com/learningequality/studio)
+ *
+ * This file must be modified if:
+ *  - File structures change in the any supported projects
+ *  - A new project wishes to be covered by this process
+ *  - There is a change to how files define their i18n messages, strings
+ *    are not being extracted properly, etc
+ *
+ * This block of code will setup paths specific to each supported
+ * project and will define how their files are processed. These paths
+ * will be used to determine which globs are processed
+ *
+ */
+
+// NP - Be sure you select a name for your project here and add it to this array
+const SUPPORTED_PROJECTS = ['kolibri', 'studio'];
```

Comment:
I don't like this at all, and it means it would be difficult to use this tooling for Kolibri plugins.

We should either allow the file paths to be configured (for use in Studio and use cases we don't anticipate) or to leverage the Kolibri plugin build machinery that already exists to get these paths (which includes specifying the locale path on a per plugin basis).

My main reason for pushing back on this strongly, is that I think if this is just for Kolibri and Studio, we should not bother making generalized code and should just copy paste across the repos. My intention was that this tooling would be useful for Kolibri plugins also and hence more generally applicable.

Reply:
Right on - on our next pass I look forward to hashing this out together. If using the build machinery still basically gets us those glob paths and we read the files in and make ASTs from the string of the file then what I'd consider the most complicated parts of the script won't require much change.

I did try to give relatively easy ways to extend this process to other projects - but it was based on an incorrect assumption about how I should break down the strings in Kolibri with regard to the plugins within each of the `kolibri/plugins/*`.

Thanks for the thorough breakdown here!